build/build.ps1

```diff
@@ -0,0 +1,223 @@
+    if (-Not (Test-Path $openCoverNuPkgOutFile)) {
+        Invoke-WebRequest `
+            -Uri "https://ci.appveyor.com/api/buildjobs/upad53qdyo1iv382/artifacts/main%2Fbin%2Fpackages%2Fnuget%2Fopencover%2FOpenCover.4.6.712.nupkg" `
+            -OutFile $openCoverNuPkgOutFile
+    }
+
+    & $buildFolder\nuget.exe install OpenCover -version $openCoverVersion -SolutionDirectory $buildFolder -source $buildFolder
+
+    $openCoverConsole = $buildFolder + 'packages/' + 'OpenCover.' + $openCoverVersion + '\tools\OpenCover.Console.exe'
+    $target = '-target:C:\Program Files\dotnet\dotnet.exe'
+    $targetArgs = '-targetargs: test ' + $testProject + ' -f ' + $runtime
+    $filter = '-filter:+[' + $appNamespace + '*]* -[' + $testNamespace + ']*'
+    $output = '-output:' + $coverageFile
+
+    & $openCoverConsole $target $targetArgs $filter $output '-register:user' '-oldStyle'
+
+    if (-not $?)
+    {
+        throw "Unit tests failed."
+    }
+
+    if (-Not (Test-Path $coverageFile))
+    {
+        return
+    }
+    if ([bool]$codeCovSecret)
+    {
+        $ENV:PATH = 'C:\\Python34;C:\\Python34\\Scripts;' + $ENV:PATH
+        python -m pip install --upgrade pip
+        pip install git+git://github.com/codecov/codecov-python.git
+        codecov -f $coverageFile -t $codeCovSecret -X gcov   
+    }
+    else
+    {
+        $reportGeneratorVersion = '2.5.7'
+        & $buildFolder\nuget.exe install ReportGenerator -version $reportGeneratorVersion -SolutionDirectory $buildFolder
+        $reportGenerator = $buildFolder + 'packages\' + 'ReportGenerator.' + $reportGeneratorVersion + '\tools\ReportGenerator.exe'
+        $targetDirectory = $buildFolder + 'OpenCoverReport\'
+        & $reportGenerator -reports:$coverageFile -targetdir:$targetDirectory
+    }
+}
+
+function CopyArtifacts
+{
+    if (-Not $isAppveyor)
+    {
+        return
+    }
+    New-Item -ItemType Directory -Force -Path $artifactsFolder | Out-Null
+    MSBuild.exe $appProject /t:pack /p:Configuration=$configuration /p:Platform=$platform /p:PackageOutputPath=$artifactsFolder /verbosity:minimal
+    if (Test-Path $coverageFile)
+    {
+        Copy-Item $coverageFile $artifactsFolder
+    }
+}
+
+function Delete-AzureResources
+{
+    Write-Host "Deleting Azure resources"
+
+    $password = ConvertTo-SecureString -AsPlainText -Force $clientSecret
+    $credentials = New-Object `
+        -TypeName System.Management.Automation.PSCredential `
+        -ArgumentList $appId, $password
+
+    Remove-AzureRmResourceGroup -Name $resourceGroupName -WarningAction SilentlyContinue -Force | Out-Null
+
+    Write-Host "Completed deleting Azure resources"
+}
+
+Build-Solution
+if (-Not $canDeploy -and -Not [bool][Environment]::GetEnvironmentVariable($connectionStringVariableName)) {
+    return
+}
+try {
+    if ($canDeploy -and -not [bool][Environment]::GetEnvironmentVariable($connectionStringVariableName)) {
+        Deploy-AzureResources
```

Comment:
>Deploy-AzureResources [](start = 8, length = 21)

for now, we dont need to deploy anything since everything can be done through mocks.